app/controllers/api/solutions/export_controller.rb

```diff
@@ -0,0 +1,24 @@
+module API
+  module Solutions
+    class ExportController < BaseController
+      before_action :use_exercise
+
+      def index
+        return render_403(:solutions_export_not_accessible) unless current_user.maintainer? || current_user.admin?
+
+        zip_file_stream = Exercise::ExportSolutionsToZipFile.(@exercise)
+
+        send_data zip_file_stream, type: 'application/zip', filename: 'solutions.zip'
+      end
+
+      private
+      def use_exercise
+        @track = Track.find_by(slug: params[:track_slug])
+        return render_track_not_found if @track.blank?
```

Comment:
I'd have a slight preference here to having a `use_track` method which is:

```ruby
def use_track
  @track = Track.find(params[:track_slug])
rescue ActiveRecord::RecordNotFound
  render_track_not_found
end
```

and then the same for exercise